docs/source/cuda_ann_benchmarks.md

```diff
@@ -60,6 +60,143 @@ There are 4 general steps to running the benchmarks:
+```
+`executable` : specifies the binary that will build/search the index. It is assumed to be
+available in `raft/cpp/build/`.
+`disabled` : denotes whether an algorithm should be excluded from benchmark runs.
+
+The usage of the script `scripts/run.py` is:
+```bash
+usage: run.py [-h] [--dataset DATASET] [--algorithms ALGORITHMS] [--indices INDICES] [--force]
+
+options:
+  -h, --help            show this help message and exit
+  --dataset DATASET     the dataset to load training points from (default: glove-100-inner)
+  --algorithms ALGORITHMS
+                        run only comma separated list of named algorithms (default: None)
+  --indices INDICES     run only comma separated list of named indices. parameter `algorithms` is ignored (default: None)
+  --force               re-run algorithms even if their results already exist (default: False)
+```
+
+Both parameters `indices` and `algorithms` ensure that the algorithm specified for an index 
+is available in `algos.yaml` and not disabled, as well as having an associated executable.
+
+#### Step 3: Evaluating Results
+The script `scripts/data_export.py` will evaluate results for a dataset whose index has been built
+and search with at least one algorithm. For every result file that is supplied to the script, the output
+will be combined and written to a CSV file.
+
+The usage of this script is:
+```bash
+usage: data_export.py [-h] --output OUTPUT [--recompute] --groundtruth GROUNDTRUTH <result_filepaths>
+
+options:
+  -h, --help            show this help message and exit
+  --output OUTPUT       Path to the output file (default: None)
+  --recompute           Recompute metrics (default: False)
+  --groundtruth GROUNDTRUTH
+                        Dataset whose groundtruth is used (default: None)
+```
+
+`result_filepaths` : whitespace separate list of result files/directories that can be capture via pattern match. For more [information and examples](#result-filepath-example)
+
+#### Step 4: Plot Results
+The script `scripts/plot.py` will plot all results evaluated to a CSV file for a given dataset.
+
+The usage of this script is:
+```bash
+usage: plot.py [-h] --result_filepath RESULT_FILEPATH [--output OUTPUT] [--x-scale X_SCALE] [--y-scale {linear,log,symlog,logit}] [--raw]
+
+options:
+  -h, --help            show this help message and exit
+  --result_filepath RESULT_FILEPATH
+                        Path to CSV Results (default: None)
+  --output OUTPUT       Path to the PNG output file (default: None)
+  --x-scale X_SCALE     Scale to use when drawing the X-axis. Typically linear, logit or a2 (default: linear)
+  --y-scale {linear,log,symlog,logit}
+                        Scale to use when drawing the Y-axis (default: linear)
+  --raw                 Show raw results (not just Pareto frontier) in faded colours (default: False)
+```
+
+All algorithms present in the CSV file supplied to this script with parameter `result_filepath`
+will appear in the plot.
+
+### bash
```

Comment:
I think we should move the `bash` instructions out to another markdown file and link from here (and maybe choose a name that makes it more clear the users could also call the underlying benchmark executables directly). We should probably mention that the Python scripts are lightweight wrappers around lower-level executables and then provide the link to those instructions.

Reply:
All `bash` content meaning those for the executables as well?

Reply:
I personally think it would be okay to just remove the instructions for the low level binaries altogether. But then that would leave to more confusion. This is an effort to make a nice clean "10 minute to benchmarking" guide. Another alternative maybe could be to isolate the python bits somewhere else? Im just really trying to make these as user friendly and easy to follow as possible. There's a lot of steps and text in the guide as it is currently (was already the case before but now even more so)

Reply:
How about this:

1. One markdown for build instructions, since they will be common for python/bash
2. One markdown for python
3. One markdown for bash

2/3 link to 1. And, 2/3 also keep their bash examples in their own markdown